lib/java/src/org/apache/thrift/transport/TFastFramedTransport.java

```diff
@@ -164,6 +166,10 @@ public void consumeBuffer(int len) {
     readBuffer.consumeBuffer(len);
   }
 
+  public void clear() {
+    readBuffer = new AutoExpandingBufferReadTransport(initialBufferCapacity, 1.5);
```

Comment:
It would be nice to share the initialization code on line 110 above in the constructor - consider factoring out a factory or calling reset from the constructor to initialize `readBuffer` for the 1st time.